backend/danswer/server/query_and_chat/query_backend.py

```diff
@@ -136,6 +142,109 @@ def query_validation(
     return QueryValidationResponse(reasoning=reasoning, answerable=answerable)
 
 
+@basic_router.get("/user-searches")
+def get_user_query_sessions(
+    user: User | None = Depends(current_user),
+    db_session: Session = Depends(get_session),
+) -> ChatSessionsResponse:
+    user_id = user.id if user is not None else None
+
+    try:
+        search_sessions = get_chat_sessions_by_user(
+            user_id=user_id, deleted=False, db_session=db_session, only_one_shot=True
+        )
+    except ValueError:
+        raise HTTPException(
+            status_code=404, detail="Chat session does not exist or has been deleted"
+        )
+
+    search_session_ids = [chat.id for chat in search_sessions]
+    first_messages = get_first_messages_for_chat_sessions(
+        search_session_ids, db_session
+    )
+    first_messages_dict = dict(first_messages)
+
+    response = ChatSessionsResponse(
+        sessions=[
+            ChatSessionDetails(
+                id=search.id,
+                name=first_messages_dict.get(search.id, search.description),
+                persona_id=search.persona_id,
+                time_created=search.time_created.isoformat(),
+                shared_status=search.shared_status,
+                folder_id=search.folder_id,
+                current_alternate_model=search.current_alternate_model,
+            )
+            for search in search_sessions
+        ]
+    )
+    return response
+
+
+@basic_router.get("/search-session/{session_id}")
+def get_search_session(
+    session_id: int,
+    is_shared: bool = False,
+    user: User | None = Depends(current_user),
+    db_session: Session = Depends(get_session),
+) -> SearchSessionDetailResponse:
+    user_id = user.id if user is not None else None
+
+    try:
+        search_session = get_chat_session_by_id(
+            chat_session_id=session_id,
+            user_id=user_id,
+            db_session=db_session,
+            is_shared=is_shared,
+        )
+    except ValueError:
+        raise ValueError("Search session does not exist or has been deleted")
+
+    if search_session.user_id is None and user_id is not None:
```

Comment:
What is this one about?

Reply:
Removing- Historical artifact! Was experimenting on a second branch with merging the chat/search flows and that check is needed for chats (due to seeding of first message)